src/Client.php

```diff
@@ -457,6 +447,28 @@ public function watch(array $pipeline = [], array $options = [])
         return $operation->execute($server);
     }
 
+    private function formatEncryptionOptions(array $options): array
+    {
+        if (isset($options['keyVaultClient'])) {
+            if ($options['keyVaultClient'] instanceof self) {
+                $options['keyVaultClient'] = $options['keyVaultClient']->manager;
+            } elseif (! $options['keyVaultClient'] instanceof Manager) {
+                throw InvalidArgumentException::invalidType('"keyVaultClient" option', $options['keyVaultClient'], [self::class, Manager::class]);
+            }
+        }
+
+        // The server requires an empty document for automatic credentials.
+        if (isset($options['kmsProviders']) && is_array($options['kmsProviders'])) {
+            foreach ($options['kmsProviders'] as $name => $provider) {
+                if ($provider === []) {
+                    $options['kmsProviders'][$name] = Document::fromPHP([]);
```

Comment:
Any reason to use `Document::fromPHP([])` instead of `(object) []` or `new stdClass`? PHPC is going to end up [encoding this as BSON anyway](https://github.com/mongodb/mongo-php-driver/blob/9eff31bd994a988743044a6cad727e3ea3a9f619/src/phongo_client.c#L1249).

Reply:
I have no idea what's the more efficient. Changed to `new stdClass`